grpcx/deamon.go

```diff
@@ -0,0 +1,84 @@
+package grpcx
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"google.golang.org/grpc"
+
+	"github.com/socialpoint-labs/bsk/contextx"
+)
+
+// Daemon represents a runnable gRPC daemon
+type Daemon struct {
+	port int
+	lis  net.Listener
+	svr  *grpc.Server
+	ef   func(error)
+	apps []Application
+}
+
+// Application is a contextx.Runner that can register gRPC services
+type Application interface {
+	contextx.Runner
+	RegisterGRPC(grpc.ServiceRegistrar)
+}
+
+// NewDaemonWithPort creates a gRPC Daemon listening at the given port.
+func NewDaemonWithPort(port int, svr *grpc.Server, ef func(error), apps ...Application) Daemon {
```

Comment:
What do you think instead of have it port, and svr and listener (on the another construct) as paremeters, have an DTO and only take what do you need, with nullable params, so if the param is nil so don't you use it, and only have a `NewDaemon` method?

Reply:
In the end I added options for everything but the gRPC server.